switch to JS API

mathieudutour opened the pull request.

mathieudutour committed:
switch to JS API

mathieudutour committed:
remove ds_store

mathieudutour committed:
remove extra files

eduardogr commented:
Hi! Nice work!

What do you think about add an entry in the .gitignore for the file ".vscode/settings.json"??

mathieudutour committed:
ignore .vscode/settings.json

mathieudutour committed:
handle finding default library styles

Pausansol committed:
General changes for v53

- Analytics module
- UI COSAlertWindow
- Improved the way to access sharedStyles in overrides coming from a library

mathieudutour commented on src/script.js:
```diff
@@ -3,25 +3,46 @@ import googleAnalytics from './analytics'
+export function getOptionSelected(libraries) {
+  
+  const options = [];
+  var optionSelected = NSComboBox.alloc().initWithFrame(NSMakeRect(0, 0, 200, 25));
```
why would you go back to using NSComboBox and COSAlertWindow?

Pausansol replied:
I want to add a pair of radio buttons in the next release for "apply to whole document" and "apply only to selection". Didn't find a way to do this with API, looked at the API documentation. If there is a way, I will be happy to change it.

mathieudutour replied:
I see

mathieudutour commented on src/script.js:
```diff
@@ -3,25 +3,46 @@ import googleAnalytics from './analytics'
-    googleAnalytics('Camilo replacement with', chosenLibrary.name, 'Library')
-
-    syncLibrary(doc, chosenLibrary)
+export function getOptionSelected(libraries) {
```
that shouldn't be exported

Pausansol replied:
You're right

mathieudutour commented on src/script.js:
```diff
@@ -3,25 +3,46 @@ import googleAnalytics from './analytics'
+export function getOptionSelected(libraries) {
+  
+  const options = [];
+  var optionSelected = NSComboBox.alloc().initWithFrame(NSMakeRect(0, 0, 200, 25));
```
let's try to keep a consistent coding style: use `const` instead of `var`, don't put `;`, etc. Eslint and Prettier should help with that, try to install plugins for them in your code editor

Pausansol replied:
I really appreciate these comments, ok, thanks!

mathieudutour commented on src/script.js:
```diff
@@ -3,25 +3,46 @@ import googleAnalytics from './analytics'
+  if (alert.runModal() == NSAlertFirstButtonReturn) {
+    var chosenLibrary = alert.viewAtIndex(0).stringValue();
+    libraries.forEach(function (lib) {
+      if (lib.name == chosenLibrary) {
```
that'd be better written as
```js
const chosenLibraryName = String(alert.viewAtIndex(0).stringValue())
const lib = libraries.find(l => l.name === chosenLibraryName)
```

Pausansol replied:
Nice, I will change, not familiar yet with arrow functions.

mathieudutour commented on src/replace-shared-styles.js:
```diff
@@ -1,39 +1,25 @@
     if (currentSharedStyle) {
-      const imported = librarySharedStyle.import()
-      // if the shared style is coming from a library, then we just want to:
-      // - import the matching style
```
any reason to remove comments instead of adding more? 😄

Pausansol replied:
mmm true, I will get them back ;)

mathieudutour commented on src/analytics.js:
```diff
@@ -1,59 +1,41 @@
+  var trackingID = "UA-128191866-1";
+  var uuidKey = "google.analytics.uuid";
+  var url = "https://www.google-analytics.com/collect?v=1";
+  var uuid = NSUserDefaults.standardUserDefaults().objectForKey(uuidKey);
```
why all this change? at the very least, this should be done outside the function so that it's executed only once

Pausansol replied:
Don't know why, but analytics was counting a new user per event. So I rolled it back to the previous working version.

Pausansol committed:
Code improvements

Pausansol committed:
Code consistency

Pausansol committed:
relive the comments
